### PR TITLE
Remove Painless Type from e-nodes in favor of Java Class

### PR DESCRIPTION
**Pull request by jdconrad**
This is the fourth part in a series to remove Painless Type.  This replaces Painless Type with Java Class for all expression type nodes in Painless.  This is a somewhat large change because the base node for all expression nodes is modified and can't really be broken up.

### CODE REVIEW

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/Definition.java [hunk 1]
```diff
@@ -618,8 +618,8 @@ public Type ClassToType(Class<?> clazz) {
         return getType(structsMap.get(ClassToName(clazz)), 0);
     }
 
-    public static Class<?> TypeToClass (Type type) {
-        if (type.dynamic) {
+    public static Class<?> TypeToClass(Type type) {
+        if ("def".equals(type.struct.name)) {
```
Comment (rjernst) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/Definition.java]: Maybe use `def.class.getShortName()`?

**Review by rjernst — APPROVED** [review]
LGTM

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/Definition.java (hunk 1 above)
Comment (jdconrad) [reply modules/lang-painless/src/main/java/org/elasticsearch/painless/Definition.java]: Updated for all of Definition.

**Comment by jdconrad** [comment]
@rjernst Thanks for the review.  Will commit as soon as CI completes.

**Pull request merged**